cloakcoin-qt.pro

```diff
@@ -1,150 +1,268 @@
+# * Optional Dependencies: turn them on or off through qmake arguments. See README.md for more explanations
+# * Security flags: various compiler flags to improve security
+# * Other compiler flags: various compiler flags, explained
+# * Translations: build the translations
+# * Files to compile: the list of all files that have to be compiled
+# * Misc: things that don't really fit anywhere else
+
+
+########################
+# QT configuration
+########################
+
+# QT modules we use. Widgets are since QT 4, but we don't support under that anyway
+QT += core gui network widgets
 TEMPLATE = app
-TARGET = cloakcoin-qt
-VERSION = 0.7.2
-INCLUDEPATH += src src/json src/qt src/tor
-DEFINES += QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE BOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN __NO_SYSTEM_INCLUDES
-#DEFINES += CURL_STATICLIB
-CONFIG += no_include_pwd
-CONFIG += thread
-CONFIG += static
 
-message($$QMAKESPEC)
+# The output file name. The extension depends on the OS
+TARGET = cloakcoin-qt
 
-#PRECOMPILED_HEADER = src/stable.h
+# Current app version
+VERSION = 2.1.0
 
-#DEFINES += DEBUG_LOCKCONTENTION DEBUG_LOCKORDER
+# We should aim to have a clean up-to-date build
+# Emit warnings for deprecations. 
+#DEFINES += QT_DEPRECATED_WARNINGS
 
-#CONFIG += console
-# UNCOMMENT THIS TO GET CONSOLE QDEBUG OUTPUT
-# CONFIG += console
+# disables all the APIs deprecated before Qt 5.6
+#DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0x050600
 
+# But that's not the case yet, so avoid disabling deprecated for now
 greaterThan(QT_MAJOR_VERSION, 4) {
     DEFINES += QT_DISABLE_DEPRECATED_BEFORE=0
 }
 
+# Is that actually necessary?
+DEFINES += QT_GUI 
+
+CONFIG += no_include_pwd thread static c++11
+INCLUDEPATH += src src/json src/qt src/tor
+
+# "Other files" to show in Qt Creator
+OTHER_FILES += \
+    doc/*.rst doc/*.txt doc/README \
+    README.md \
+    res/bitcoin-qt.rc \
+    src/test/*.cpp src/test/*.h src/qt/test/*.cpp src/qt/test/*.h \
+    src/qt/locale/*.ts
+
+########################
+# Debug variables
+########################
+
+# UNCOMMENT THIS TO GET CONSOLE QDEBUG OUTPUT
+#CONFIG += console
+
+#DEFINES += DEBUG_LOCKCONTENTION
+#DEFINES += DEBUG_LOCKORDER
+
 #DEFINES += ENABLE_WALLET
 
+########################
+# Where to compile
+########################
+
 OBJECTS_DIR = build
 MOC_DIR = build
 UI_DIR = build
+RCC_DIR = build
 
-message(LIBS = $$LIBS)
+########################
+# Platform defines
+########################
+
+win32:DEFINES += WIN32
+linux:DEFINES += LINUX
+macx:DEFINES += MAC_OSX
+
+
+########################
+# Mandatory Dependencies
+########################
+
+# Configure boost
+DEFINES += BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE BOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN
+
+# What's that? The only Google match I found is form Berkely DB. We might not need that
+DEFINES +=  __NO_SYSTEM_INCLUDES
```

Comment:
This one instructs the compiler/linker to not look at default system include paths, for gcc it would be equivalent as `-nostdinc` flag.